src/webgpu/api/validation/queue/buffer_mapped.spec.ts

```diff
@@ -1,47 +1,261 @@
 export const description = `
-Tests for map-state of mappable buffers used in submitted command buffers.
+Validation tests for the map-state of mappable buffers used in submitted command buffers.
 
-- x= just before queue op, buffer in {BufferMapStatesToTest}
-- x= in every possible place for mappable buffer:
-  {submit, writeBuffer, copyB2B {src,dst}, copyB2T, copyT2B, ..?}
+Tests every operation that has a dependency on a buffer
+  - writeBuffer
+  - copyB2B {src,dst}
+  - copyB2T
+  - copyT2B
 
-TODO: generalize existing test
+Test those operations against buffers in the following states:
+  - Unmapped
+  - In the process of mapping
+  - mapped
+  - mapped with a mapped range queried
+  - unmapped after mapping
+  - mapped at creation
+
+Also tests every order of operations combination of mapping operations and command recording
+operations to ensure the mapping state is only considered when a command buffer is submitted.
 `;
 
 import { makeTestGroup } from '../../../../common/framework/test_group.js';
 import { ValidationTest } from '../validation_test.js';
 
-export const g = makeTestGroup(ValidationTest);
+class F extends ValidationTest {
+  async runBufferDependencyTest(
+    options: {
+      usage: number;
+      mapMode: number;
+      bufferSize?: number;
+    },
+    callback: Function
+  ): Promise<void> {
+    const { usage, mapMode, bufferSize } = options;
+
+    const bufferDesc = {
+      usage,
+      size: bufferSize || 8,
+      mappedAtCreation: false,
+    };
+
+    // Create a mappable buffer, and one that will remain unmapped for comparison.
+    const mappableBuffer = this.device.createBuffer(bufferDesc);
+    const unmappedBuffer = this.device.createBuffer(bufferDesc);
```

Comment:
hmm do you think we need this buffer as well? I guess it checks state of one buffer doesn't affect another?

Reply:
That was my intent, yeah. It's a bit of a edge case, you certainly wouldn't expect anything like that to happen, but I figured it might help catch something like, say, issues in an implementation that does buffer pooling under the hood.